packages/node_modules/pouchdb-adapter-http/src/index.js

```diff
@@ -169,7 +166,10 @@ function HttpPouch(opts, callback) {
   function ajax(userOpts, options, callback) {
     var reqAjax = userOpts.ajax || {};
     var reqOpts = assign(clone(ajaxOpts), reqAjax, options);
-    log(reqOpts.method + ' ' + reqOpts.url);
+    /* istanbul ignore if */
+    if (api.constructor.listeners('debug').length) {
```

Comment:
this `if` check is to avoid even the array allocation in the case where nobody is actually listening in on the `'debug'` event.